docs/APIListToBePorted.md

```diff
@@ -0,0 +1,207 @@
+16. up_rtc_initialize()
+
+## API Description
+
+- Add low level chip specific API **up_flashinitialize()** to create an initialize MTD device instance. MTD devices are not registered in the file system, but are created as instances that can be bound to other functions (such as a block or character driver front end).
+```c
+ * Prototype: struct mtd_dev_s *up_flashinitialize(void)
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   Valid MTD device structure reference on success; a NULL on failure
+```
+*File Location:*	"os/arch/arm/src/chip/chip_flash.c"
+
+*Prototype Location:*	"os/include/tinyara/fs/mtd.h"
+
+- Add low level chip specific API **up_idle()** to be executed when their is no other ready-to-run task. This is processor idle time and will continue until some interrupt occurs to cause a context switch from the idle task. Processing in this state may be processor-specific. e.g., this is where power management operations might be performed.
+
+```c
+ * Prototype: void up_idle(void)
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   none
+```
+To this we can further add **up_idlepm()** to perform IDLE state power management.
+```c
+static void up_idlepm(void)
+```
+*File Location:*	"os/arch/arm/src/chip/chip_idle.c"
+
+*Prototype Location:*	"os/include/tinyara/arch.h"
+
+- Add low level chip specific API **up_enable_irq()** to enable the IRQ specified by 'irq'
+```c
+ * Prototype: void up_enable_irq(int irq)
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   none
+```
+*File Location:*	"os/arch/arm/src/chip/chip_irq.c"
+
+*Prototype Location:*	"os/include/tinyara/arch.h"
+
+- Add low level chip specific API **up_netinitialize()** to initialize network domain
+[TO DO]
+```c
+ * Prototype: void up_netinitialize(void)
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   none
+```
+*File Location:*	"os/arch/arm/src/chip/chip_enet.c"
+
+*Prototype Location:*	"os/arch/arm/src/common/up_internal.h"
+
+- Add low level chip specific API's to initialize and register serial console and ports:
+
+a. **up_serialinit()** : Register serial console and serial ports.
+```c
+ * Prototype: void up_serialinit(void)
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   none
+```
+b. **up_earlyserialinit()** : Performs the low level UART initialization early in debug so that
+the serial console will be available during bootup.  This must be called before up_serialinit.
+```c
+ * Prototype: void up_earlyserialinit(void)
+ * Input Parameters:
+ *   none
+ * Returned Value:
+ *   none
+```
+*File Location:*	"os/arch/arm/src/chip/chip_serial.c"
+
+*Prototype Location:*	""
+
+- Add low level chip specific API's to receive and transfer characters through UART:
+
+a. **up_putc()** : Output one byte on the serial console
+```c
+ * Prototype: int up_putc(int ch)
+ * Input Parameters:
+ *   ch - chatacter to output
+ * Returned Value:
+ *   sent character
+```
+b. **up_getc()** : Read one byte from the serial console
+```
```

Comment:
We'd better marking this box as c language like line 107.

Reply:
done